Set CMP0167 for CMake 3.30

https://cmake.org/cmake/help/latest/policy/CMP0167.html

```
  CMake Error (dev) at CMakeLists.txt:123 (find_package):
    Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
    --help-policy CMP0167" for policy details.  Use the cmake_policy command to
    set the policy and suppress this warning.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,10 @@ cmake_policy(VERSION 3.15...3.27)
 if(POLICY CMP0144)
   cmake_policy(SET CMP0144 NEW)
 endif(POLICY CMP0144)
+# cmake >= 3.30
+if(POLICY CMP0167)
+  cmake_policy(SET CMP0167 NEW)
+endif(POLICY CMP0167)
 
 project(amici)
 

--- a/swig/CMakeLists_model.cmake
+++ b/swig/CMakeLists_model.cmake
@@ -1,4 +1,15 @@
 cmake_minimum_required(VERSION 3.15)
+cmake_policy(VERSION 3.15...3.27)
+
+# cmake >=3.27
+if(POLICY CMP0144)
+  cmake_policy(SET CMP0144 NEW)
+endif(POLICY CMP0144)
+# cmake >= 3.30
+if(POLICY CMP0167)
+  cmake_policy(SET CMP0167 NEW)
+endif(POLICY CMP0167)
+
 
 if(DEFINED ENV{SWIG})
   set(SWIG_EXECUTABLE $ENV{SWIG})

